### ISSUE TITLE
Track contamination in new leaderboard

### ISSUE THREAD

**Issue by Muennighoff** [embeddings-benchmark/mteb#1556]
Just discussed this with @sivareddyg & others, but I think once we have this, many people would be willing to help us do all the annotations. Ideally it is super easy to add contamination annotations via a simple PR

- [x] Add easy way to add training data metadata of models to
- [ ] Modify UI to select whether contaminated models are allowed or not
- [ ] Modify UI to maybe always display a small symbol next to numbers which are contaminated based on the metadata
- [ ] Maybe add small section [here](https://github.com/embeddings-benchmark/mteb?tab=readme-ov-file#usage-documentation) on how to add contamination annotations

**Comment by KennethEnevoldsen** [comment]
So you are suggesting to replace:
```
class ModelMeta(BaseModel):
    name: str | None
    revision: str | None
    ...
    zero_shot_benchmarks: list[str] | None = None
    citation: str | None = None
```

with 

```
class ModelMeta(BaseModel):
    name: str | None
    revision: str | None
    ...
    training_datasets: list[str] | None = None
    citation: str | None = None
```

And rely on users to annotate it? I would be up for that (@x-tabdeveloping this is also more in line with your idea)

We can extrapolate `zero_shot_benchmarks` from training datasets. What do we do for models where the training data is not known?

UI is pretty simple once we have it

**Comment by x-tabdeveloping** [comment]
Yeah, once we've implemented the logic I can add it to the UI

**Comment by isaac-chung** [comment]
Does that mean that #1372 is obsolete? If so we can close it.

**Comment by KennethEnevoldsen** [comment]
Added a PR

**Comment by Muennighoff** [comment]
Here's an example for displaying contamination: https://huggingface.co/spaces/allenai/reward-bench though we probably want to be more finegrained

**Comment by x-tabdeveloping** [comment]
Thanks for implementing the metadata attribute for this.
I have a number of questions and concerns relating to this.

### 1. We have limited information
As it currently stands, only 18 of the 187 models that have associated metadata in MTEB have any annotation regarding what they were trained on, and I suspect that some of these are quite incomplete.

I personally don't think it makes a lot of sense to implement a filter in the leaderboard until we add this data on more models.

I'm also wondering who should be made responsible for doing this. What happens with models, for which this information is not published? Can we push people to give us information about their training data?

### 2. Indirect contamination
It might be the case that certain datasets that were used for training a model contain examples from another dataset's test or dev split, or contain the other dataset altogether. What are we planning to do in such cases, do we ignore this?

Some training datasets might be generated with LLMs, that might have seen the original source. Do we care about this?

### 3. Pretraining datasets
Models like e5-mistral are based on a model that was pretrained on vast amounts of data, and is likely to have seen some of the test data in raw form from certain datasets.
Is this something we consider to be problematic, and if so how are we planning to account for this?

**Comment by KennethEnevoldsen** [comment]
1) 
> I personally don't think it makes a lot of sense to implement a filter in the leaderboard until we add this data on more models.

Let us add the filter that removed model that know are not zero-shot (then people will add the annotations as it goes)

2)
> Do we care about this? 

Not now

3)
> Is this something we consider to be problematic, and if so how are we planning to account for this?

It would also be leakage if e.g. test data is included.

**Comment by KennethEnevoldsen** [comment]
I believe this is essentially done - we can open individual issue for future problems
